library/bignum_mod.c

```diff
@@ -176,6 +176,28 @@ int mbedtls_mpi_mod_modulus_setup( mbedtls_mpi_mod_modulus *m,
 
 /* BEGIN MERGE SLOT 2 */
 
+int mbedtls_mpi_mod_mul( mbedtls_mpi_mod_residue *X,
+                         const mbedtls_mpi_mod_residue *A,
+                         const mbedtls_mpi_mod_residue *B,
+                         const mbedtls_mpi_mod_modulus *N )
+{
+    if( N->limbs == 0 )
+        return MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+
+    if( X->limbs != N->limbs || A->limbs != N->limbs || B->limbs != N->limbs )
+        return MBEDTLS_ERR_MPI_BAD_INPUT_DATA;
+
+    mbedtls_mpi_uint *T = mbedtls_calloc( N->limbs * 2 + 1, ciL );
+    if( !T )
```

Comment:
(very minor; only change if something else needs to) While my personal preference is for `!` here (the idea of "not" for "failure" makes the code very easy to read) the bignum code returning `MBEDTLS_ERR_MPI_ALLOC_FAILED` seems to always use `== NULL`

Reply:
That is also the safest approach secure coding wise too. The `== NULL` is clearly defined behavior while "if(!*X)" is implementation defined

Reply:
Err, `if (!p)` and `if (p == NULL)` are equivalent for a pointer `p`.

Reply:
The code base makes conversions to boolean explicit almost everywhere. I thought this was mandated by the coding standards, but it isn't. It should probably be, because it's prevalent enough to be considered an implicit rule.